common/dynamicconfig/constants.go

```diff
@@ -2707,25 +2708,25 @@ var IntKeys = map[IntKey]DynamicInt{
 		KeyName:      "limit.historySize.error",
 		Filters:      []Filter{DomainName},
 		Description:  "HistorySizeLimitError is the per workflow execution history size limit",
-		DefaultValue: 200 * 1024 * 1024,
+		DefaultValue: 50 * 1024 * 1024,
 	},
 	HistorySizeLimitWarn: DynamicInt{
 		KeyName:      "limit.historySize.warn",
 		Filters:      []Filter{DomainName},
 		Description:  "HistorySizeLimitWarn is the per workflow execution history size limit for warning",
-		DefaultValue: 50 * 1024 * 1024,
+		DefaultValue: 25 * 1024 * 1024,
 	},
 	HistoryCountLimitError: DynamicInt{
 		KeyName:      "limit.historyCount.error",
 		Filters:      []Filter{DomainName},
 		Description:  "HistoryCountLimitError is the per workflow execution history event count limit",
-		DefaultValue: 200 * 1024,
+		DefaultValue: 50000,
```

Comment:
I would worry that this would be a bit disruptive to get more restrictivre config during an upgrade - its' a bit of a breaking change which would likely affect customers. though I agree it's probably too high to be safe for a lot of high throughput prod envs. 

For our own internal use I have some ideas (will discuss offline) about handling our config defaults internally. I am not sure open-source customers would welcome having their workflows hit errors on upgrade however.